pkg/terraform/exec/plugins/plugin_versions_generate.go

```diff
@@ -0,0 +1,249 @@
+// +build tools
+
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"text/template"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing"
+	"golang.org/x/mod/modfile"
+	"golang.org/x/mod/module"
+)
+
+type TFPluginInfo struct {
+	Module  module.Version
+	Name    string
+	Tag     string
+	Version string
+}
+
+type TFPlugins struct {
+	Pkg          string
+	TFPluginInfo []TFPluginInfo
+}
+
+func getTemplate() string {
+	return `// Code generated by plugin_versions_generate.go; DO NOT EDIT.
+
+package {{ .Pkg }}
+
+{{ with .TFPluginInfo }}
+        {{- range . }}
+
+func Get{{ .Name }}Version() string {
+	return "{{ .Version }}"
+}
+
+        {{- end }}
+{{- end}}
+`
+}
+
+// getTerraformPluginInfo finds terraform providers used in go.mod, does any
+// necessary replacements and saves into our plugin map.
+func getTerraformPluginInfo(file *modfile.File, plugins *map[string]*TFPluginInfo) {
+
+	// terraform provider and name regex
+	re := regexp.MustCompile(`^[[:alnum:]][[:alnum:]\.\-/]+(terraform-provider-([[:alnum:]]+))`)
```

Comment:
I've updated this to use your recommended regex. While it may not seem too obvious, my regular expression did indeed ensure the slash was there before terraform-provider- ;-) But I agree, yours is not as restrictive.

Reply:
Yours **permitted** a slash but did not **enforce** a slash.
https://play.golang.org/p/-SUxddgH7gv